cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -124,6 +129,64 @@ public long size() throws IOException {
   }
 
   /**
+   * Do some processing on locations contained in the {@code locationDir}, using the {@code processor}.
+   *
+   * @param locationDir location to start iterating from
+   * @param recursive true if this method should be called on the directory {@link Location}s found from
+   *                  {@code locationDir}
+   * @param processor used to process the locations contained by the {@code locationDir}, whether they are directories
+   *                  or not. If the {@link Processor#process} method returns false on any {@link Location} object
+   *                  processed, this method will return the current result of the processor.
+   * @param <R> Type of the return value
+   * @throws IOException if the locations could not be read
+   */
+  public static <R> R processLocations(Location locationDir, boolean recursive, Processor<LocationStatus, R> processor)
+    throws IOException {
+    Preconditions.checkArgument(locationDir.isDirectory(), "Input location has to be a directory");
+    LocationFactory locationFactory = locationDir.getLocationFactory();
+    if (locationFactory instanceof HDFSLocationFactory) {
+      // Treat the HDFS case
+      Deque<Path> pathStack = Lists.newLinkedList();
+      pathStack.push(new Path(locationDir.toURI().toString()));
```

Comment:
I think `Path` has a constructor that takes `URI`. No need to call `toString()`.